### PR TITLE
Miscellaneous improvements to `SimBeamSpotObjects` handling utilities

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a follow-up PR to https://github.com/cms-sw/cmssw/pull/41918.
Miscellaneous improvements to `SimBeamSpotObjects` handling utilities, including:
   * add expected transverse width to the payload inspector
   * add `CondCore/BeamSpotPlugins` unit tests for the exercise the plots introduced in https://github.com/cms-sw/cmssw/pull/41918
   * miscellaneous clean-up of the plugins to create and read  `SimBeamSpotObjects` payloads
   * usage of auto-generated configuration fragments from `fillDescriptions` whenever possible
   * introduce some utilities to directly dump to sqlite and consequently upload to DB all vertex smearing PSets from `IOMC/EventVertexGenerators/python/VtxSmearedParameters_cfi.py`

#### PR validation:

Relies on the existing (and augmented) unit tests.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by tvami** [comment]
+1
 * dedicated unit tests pass

**Comment by civanch** [comment]
+1

File: CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h [hunk 1]
```diff
@@ -681,14 +685,15 @@ namespace simBeamSpotPI {
...
   template <class PayloadType>
   class SimBSParamsHelper {
-    typedef std::array<double, 9> bshelpdata;
+    typedef std::array<double, 10> bshelpdata;
 
   public:
     SimBSParamsHelper(const std::shared_ptr<PayloadType>& bs) {
       // fill in the values
       m_values[0] = bs->x(), m_values[1] = bs->y(), m_values[2] = bs->z();
       m_values[3] = bs->sigmaZ(), m_values[4] = bs->betaStar(), m_values[5] = bs->emittance();
```
Comment (perrotta) [inline CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h:694]:
```suggestion
      m_values[parameters::X] = bs->x(), m_values[parameters::Y] = bs->y(), m_values[parameters::Z] = bs->z();
      m_values[parameters::sigmaZ] = bs->sigmaZ(), m_values[parameters::betaStar] = bs->betaStar(),
      m_values[parameters::emittance] = bs->emittance();
```

File: CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h [hunk 2]
```diff
@@ -681,14 +685,15 @@ namespace simBeamSpotPI {
...
-      m_values[6] = bs->phi(), m_values[7] = bs->alpha(), m_values[8] = bs->timeOffset();
+      m_values[6] = (1 / std::sqrt(2)) * std::sqrt(bs->emittance() * bs->betaStar()) * 10000.f;
+      m_values[7] = bs->phi(), m_values[8] = bs->alpha(), m_values[9] = bs->timeOffset();
```
Comment (perrotta) [inline CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h:696]:
```suggestion
      m_values[parameters::expTransWidth] = (1 / std::sqrt(2)) * std::sqrt(bs->emittance() * bs->betaStar()) * 10000.f;
      m_values[parameters::phi] = bs->phi(), m_values[parameters::alpha] = bs->alpha(),
      m_values[parameters::timeOffset] = bs->timeOffset();
```

**Review by perrotta — COMMENTED** [review]
If you use the parameters in the enum you can probably save some gymn during a possible further update

File: CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h [hunk 3]
```diff
@@ -681,14 +685,15 @@ namespace simBeamSpotPI {
    */
   template <class PayloadType>
   class SimBSParamsHelper {
-    typedef std::array<double, 9> bshelpdata;
+    typedef std::array<double, 10> bshelpdata;
```
[the remark this answers is not in the document]
Comment (mmusich) [reply CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h]: thanks, addressed in [9877ac0](https://github.com/cms-sw/cmssw/pull/41967/commits/9877ac09b974681ef5d0bf058326d23e5f0bc8e0)

File: CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h (hunk 2 above)
Comment (mmusich) [reply CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h]: thanks, addressed in [9877ac0](https://github.com/cms-sw/cmssw/pull/41967/commits/9877ac09b974681ef5d0bf058326d23e5f0bc8e0)

File: CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h (hunk 1 above)
Comment (mmusich) [reply CondCore/BeamSpotPlugins/interface/BeamSpotPayloadInspectorHelper.h]: thanks, addressed in [9877ac0](https://github.com/cms-sw/cmssw/pull/41967/commits/9877ac09b974681ef5d0bf058326d23e5f0bc8e0)

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by mmusich** [comment]
is there anything holding this up?

**Comment by perrotta** [comment]
> is there anything holding this up?

We don't merge PRs while building a (pre) release. Now pre2 is out

**Comment by perrotta** [comment]
+1

**Pull request merged**